### ISSUE TITLE
Crash on iOS at shell::GPUSurfaceGL::AcquireFrame

### ISSUE THREAD

**Issue by youyou76** [flutter/flutter#16612]
## Steps to Reproduce

iPhone 8 Plus, iOS 11.2.5

## Logs
```log
shell::GPUSurfaceGL::AcquireFrame(SkISize const&) (in Flutter) + 404

shell::GPURasterizer::DrawToSurface(flow::LayerTree&) (in Flutter) + 48

shell::GPURasterizer::DoDraw(std::__1::unique_ptr<flow::LayerTree, std::__1::default_delete<flow::LayerTree> >) (in Flutter) + 128

std::__1::__function::__func<std::__1::__bind<void (shell::GPURasterizer::*)(std::__1::unique_ptr<flow::LayerTree, std::__1::default_delete<flow::LayerTree> >), shell::GPURasterizer*, std::__1::placeholders::__ph<1> const&>, std::__1::allocator<std::__1::__bind<void (shell::GPURasterizer::*)(std::__1::unique_ptr<flow::LayerTree, std::__1::default_delete<flow::LayerTree> >), shell::GPURasterizer*, std::__1::placeholders::__ph<1> const&> >, void (std::__1::unique_ptr<flow::LayerTree, std::__1::default_delete<flow::LayerTree> >)>::operator()(std::__1::unique_ptr<flow::LayerTree, std::__1::default_delete<flow::LayerTree> >&&) (in Flutter) + 60

shell::GPURasterizer::Draw(fxl::RefPtr<flutter::Pipeline<flow::LayerTree> >) (in Flutter) + 516
std::__1::__function::__func<shell::Animator::Render(std::__1::unique_ptr<flow::LayerTree, std::__1::default_delete<flow::LayerTree> >)::$_0, std::__1::allocator<shell::Animator::Render(std::__1::unique_ptr<flow::LayerTree, std::__1::default_delete<flow::LayerTree> >)::$_0>, void ()>::operator()() (in Flutter) + 136

fml::MessageLoopImpl::RunExpiredTasks() (in Flutter) + 2428

fml::MessageLoopDarwin::OnTimerFire(__CFRunLoopTimer*, fml::MessageLoopDarwin*) (in Flutter) + 32
```

## Flutter Doctor
```terminal
[✓] Flutter (Channel unknown, v0.1.6-pre.36, on Mac OS X 10.13.1 17B1003, locale zh-Hans-CN)
    • Flutter version 0.1.6-pre.36 at /Users/kangkai/fwn_idlefish/flutter
    • Framework revision 735c8d37eb (4 days ago), 2018-04-12 01:30:22 +0800
    • Engine revision e61bb9ac3a
    • Dart version 2.0.0-dev.35.flutter-290c576264

[!] Android toolchain - develop for Android devices (Android SDK 26.0.2)
    • Android SDK at /Users/kangkai/Library/Android/sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-27, build-tools 26.0.2
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-915-b08)
    ! Some Android licenses not accepted.  To resolve this, run: flutter doctor --android-licenses

[✓] iOS toolchain - develop for iOS devices (Xcode 9.2)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 9.2, Build version 9C40b
    • ios-deploy 1.9.2
    • CocoaPods version 1.2.0

[✓] Android Studio (version 3.0)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-915-b08)
```

**Comment by eseidelGoogle** [comment]
@cbracken or @chinmaygarde would know.

**Comment by chinmaygarde** [comment]
Is there a reduced test case or steps to reproduce? These frames are very common and appear in every call to render a scene.

**Comment by Hixie** [comment]
Without steps to reproduce or some more details on this it's not clear what we can do with it.

We have other similar reports in #14178 and #14179.

**Comment by timsneath** [comment]
@youyou76, we'd love to help you out here. Is there anything else you can provide, do you think? On our side, @chinmaygarde and @Hixie, is there something they could log in their app that would help with the diagnosis?

**Comment by Hixie** [comment]
I don't think we really have enough information to even know what information to ask for, unfortunately. Even just an .ipa of the app and instructions for how to trigger it would be a good start for us.

**Comment by mjohnsullivan** [comment]
ipa and additional info's inbound in an email

**Comment by eseidelGoogle** [comment]
@chinmaygarde any comment on the robustness of viewWillAppear/viewDidDisappear?  Do we ASSERT that folks don't call those twice for example?

**Comment by eseidelGoogle** [comment]
An update from @kangwang1988 "Even we ensure the logic that the function in FlutterViewController's onSurfaceUpdated:YES/NO are called symmetrically, we still find some jailbroken devices(ios8/ios9) may experience the Acquire frame related crash."

This is the only serious iOS issue being reported by customer:gold, and I'm told is (combined with the 2 android crashers) blocking further deployment of Flutter.

@chinmaygarde we should brainstorm a little today what we can do here.

**Comment by Hixie** [comment]
@kangwang1988 Can you let us know if you still see this once you've updated to the latest engine as shipped in today's beta? Thanks.

**Comment by korbin-w** [comment]
Thanks for your response @Hixie.  kangwang1988 and our guys are doing the update now, and then we will take several days to launch the new update to some bucket users to verify.  We will response the result before May. 8th.

**Comment by Hixie** [comment]
@korbin-w Very well! Keep us posted!

**Comment by Hixie** [comment]
@korbin-w was the news good?

**Comment by kangwang1988** [comment]
@Hixie 
We've upgraded the flutter to betav0.3.1. 
Now we're testing it before releasing to the public. This issue seems disappeared, but I'm not sure if it will appear after released. Currently, the crashes we found is the same as #16718
I've cherry-picked the skia workaound 
 125729: Disable unpack-row-length support on iOS 
to make a quickfix.
We'll update the status for this issue after we've released.(Probably this week)
Thanks.
